scripts/gen-protoc-gen-dart

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+## Check dependencie(s)
+
+[[ $(command -v dart) ]] || { echo "'dart' not found!" ; dep_check="false" ;}
+[[ $(command -v wget) ]] || { echo "'wget' not found!" ; dep_check="false" ;}
+[[ $(command -v tar) ]] || { echo "'gzip' not found!" ; dep_check="false" ;}
+
+[[ ${dep_check} = "false" ]] && { echo "Some dependencie(s) isn't installed yet. Please install that dependencie(s)" ; exit 1 ;}
+
+## Variables
+protobuf_src="https://github.com/google/protobuf.dart/archive/refs/tags/protoc_plugin-19.2.0+1.tar.gz"
+setdir="$( cd "$( dirname "${BASH_SOURCE[0]}" )" &> /dev/null && pwd)" # this line powered by stackoverflow
+kernelname="$(uname -s | tr '[:upper:]' '[:lower:]' || { echo 'kernel name can not definied' ; exit 1 ;})"
+
+# Defaults
+setfile="protoc_plugin/bin/protoc_plugin.dart"
+save_file="protoc-gen-dart_${kernelname}_amd64"
+
+# Check dir else create save dir
+if [[ $(basename ${setdir}) = "scripts" ]] ; then
+    if [[ $(basename $(dirname ${setdir})) = "starport" ]] ; then
```

Comment:
```suggestion
    if [[ $(basename $(dirname "${setdir}")) = "starport" ]] ; then
```

use the double quote to prevent globbing and word splitting